src/sonic-yang-models/yang-models/sonic-dash.yang

```diff
@@ -0,0 +1,476 @@
+                    }
+                }
+
+            } /* end of list DASH_ROUTING_TYPE_LIST */
+        } /* end of container DASH_ROUTING_TYPE */
+
+        container DASH_ROUTE_TABLE {
+            description "DASH Route table (LPM) associated to an ENI for outbound routing; part of APP_DB";            
+
+            list DASH_ROUTE_TABLE_LIST {
+                key "eni prefix";
+
+                leaf eni {
+                    type leafref {
+                        path /dash:sonic-dash/dash:DASH_ENI/dash:DASH_ENI_LIST/dash:name;
+                    }
+                }
+
+                leaf prefix {
+                     type stypes:sonic-ip-prefix;
+                }
+ 
+                leaf action_type {
+                    type leafref {
+                        path /dash:sonic-dash/dash:DASH_ROUTING_TYPE/dash:DASH_ROUTING_TYPE_LIST/dash:name;
+                    }
+                }
+
+                leaf vnet {
+                    when "((current()/../action_type = 'vnet') or (current()/../action_type = 'vnet-direct'))";
+                    type leafref {
+                        path /dash:sonic-dash/dash:DASH_VNET/dash:DASH_VNET_LIST/dash:name;
+                    }
+                }
+
+                leaf appliance {
+                    when "(current()/../action_type = 'appliance')";
+                    type leafref {
+                        path /dash:sonic-dash/dash:DASH_APPLIANCE/dash:DASH_APPLIANCE_LIST/dash:name;
+                    }
+                }
+
+                leaf overlay_ip {
+                    when "((current()/../action_type = 'vnet') or (current()/../action_type = 'vnet-direct'))";
+                    description "Overlay IP to use for mapping lookup, if routing_type is vnet-direct";
+                    type inet:ip-address;
+                }
+
+                leaf overlay_sip {
+                    when "(current()/../action_type = 'servicetunnel')";
+                    description "Overlay src ip for service tunnel";
+                    type inet:ip-address;
+                }
+
+                leaf overlay_dip {
+                    when "(current()/../action_type = 'servicetunnel')";
+                    description "Overlay dst ip for service tunnel";
+                    type inet:ip-address;
+                }
+
+                leaf underlay_sip {
+                    when "(current()/../action_type = 'servicetunnel')";
+                    description "Underlay src ip for service tunnel";
+                    type inet:ip-address;
+                }
+
+                leaf underlay_dip{
+                    when "(current()/../action_type = 'servicetunnel')";
+                    description "Underlay dst ip for service tunnel";
+                    type inet:ip-address;
+                }
+
+            } /* end of list DASH_ROUTE_TABLE_LIST */
+        } /* end of container DASH_ROUTE_TABLE */
+
+        container DASH_VNET_MAPPING_TABLE {
```

Comment:
When GNMI run yang validation, GNMI needs to read current config from ApplDB and merge with incoming config.
DASH_ROUTE_TABLE and DASH_VNET_MAPPING_TABLE have large number of entries, so it will spend too much time to read these 2 tables.
My suggestion is to ignore existing DASH_ROUTE_TABLE and DASH_VNET_MAPPING_TABLE in ApplDB, therefore we should use different name pattern for DASH_ROUTE_TABLE and DASH_VNET_MAPPING_TABLE, and then GNMI can ignore these 2 tables and only read other tables. For example, DASH_I_ROUTE_TABLE and DASH_I_VNET_MAPPING_TABLE

Reply:
I don't think gnmi should be reading current config from APPDB for Mapping and Routing. Lets discuss to understand your concern.

Reply:
Maybe I can use redis api like redis.keys("DASH_") to read other DASH table except Mapping and Routing, and then Mapping and Routing table should use different name pattern.
Please let me know what you think.